tcms/rpc/tests/test_testcase.py

```diff
@@ -16,13 +21,66 @@
                                   TestPlanFactory, UserFactory, VersionFactory)
 
 
-class TestNotificationRemoveCC(APITestCase):
-    """ Tests the XML-RPC testcase.notication_remove_cc method """
+class TestValidateEmail(unittest.TestCase):
+    def test_non_list_email_collection(self):
+        with self.assertRaisesRegex(TypeError, 'cc_list should be a list object.'):
+            _validate_cc_list(('example@example.com',))
+
+    def test_invalid_email(self):
+        with self.assertRaises(ValidationError):
+            _validate_cc_list(['@example.com'])
+
+    def test_valid_email_list(self):
+        _validate_cc_list(['example@example.com'])
+
+
+class TestAddNotificationCCPermission(APIPermissionsTestCase):
+    permission_label = 'testcases.change_testcase'
+
+    def _fixture_setup(self):
+        super()._fixture_setup()
+
+        self.default_cc = 'example@example.com'
+        self.testcase = TestCaseFactory()
+
+    def verify_api_with_permission(self):
+        self.rpc_client.TestCase.add_notification_cc(self.testcase.pk, [self.default_cc])
+        self.assertEqual(len(self.testcase.emailing.get_cc_list()), 1)
+        self.assertEqual(self.testcase.emailing.get_cc_list(), [self.default_cc])
+
+    def verify_api_without_permission(self):
+        with self.assertRaisesRegex(ProtocolError, '403 Forbidden'):
+            self.rpc_client.TestCase.add_notification_cc(self.testcase.pk, [self.default_cc])
```

Comment:
I have added another class which decends from`APITestCase` to test the the corner case when `TestCase` id is invalid. Those tests were not working from descendants of `APIPermissionsTestCase`. 

Anther way to implement it would be to use `subTest` inside `verify_api_with_permission` function and test the corner case of TestCase id being invalid in one `subTest` and normal functionality with a valid `TestCase` id in another.